test/mean_function.jl

```diff
@@ -1,47 +1,53 @@
-        P = 3
-        Q = 2
-        D = 4
-        # X = ColVecs(randn(rng, D, P))
-        x = randn(P)
-        x̄ = randn(P)
-        f = ZeroMean{Float64}()
+        rng, N, D = MersenneTwister(123456), 5, 3
+        x1 = randn(rng, N)
+        xD = ColVecs(randn(rng, D, N))
+        xD′ = RowVecs(randn(rng, N, D))
 
-        for x in [x]
-            @test AbstractGPs._map_meanfunction(f, x) == zeros(size(x))
-            # differentiable_mean_function_tests(f, randn(rng, P), x)
-        end
+        m = ZeroMean{Float64}()
+
+        for x in [x1, xD, xD′]
+            @test AbstractGPs._map_meanfunction(m, x) == zeros(size(x))
+            differentiable_mean_function_tests(m, randn(rng, N), x)
 
-        # Manually verify the ChainRule. Really, this should employ FiniteDifferences, but
-        # currently ChainRulesTestUtils isn't up to handling this, so this will have to do
-        # for now.
-        y, pb = rrule(AbstractGPs._map_meanfunction, f, x)
-        @test y == AbstractGPs._map_meanfunction(f, x)
-        Δmap, Δf, Δx = pb(randn(P))
-        @test iszero(Δmap)
-        @test iszero(Δf)
-        @test iszero(Δx)
+            # Manually verify the ChainRule. Really, this should employ FiniteDifferences, but
+            # currently ChainRulesTestUtils isn't up to handling this, so this will have to do
+            # for now.
+            y, pb = rrule(AbstractGPs._map_meanfunction, m, x)
+            @test y == AbstractGPs._map_meanfunction(m, x)
+            Δmap, Δf, Δx = pb(randn(rng, N))
+            @test iszero(Δmap)
+            @test iszero(Δf)
+            @test iszero(Δx)
+        end
     end
     @testset "ConstMean" begin
-        rng, D, N = MersenneTwister(123456), 5, 3
-        # X = ColVecs(randn(rng, D, N))
-        x = randn(rng, N)
+        rng, N, D = MersenneTwister(123456), 5, 3
+        x1 = randn(rng, N)
+        xD = ColVecs(randn(rng, D, N))
+        xD′ = RowVecs(randn(rng, N, D))
+
         c = randn(rng)
         m = ConstMean(c)
 
-        for x in [x]
+        for x in [x1, xD, xD′]
             @test AbstractGPs._map_meanfunction(m, x) == fill(c, N)
-            # differentiable_mean_function_tests(m, randn(rng, N), x)
+            differentiable_mean_function_tests(m, randn(rng, N), x)
         end
     end
     @testset "CustomMean" begin
-        rng, N, D = MersenneTwister(123456), 11, 2
-        x = randn(rng, N)
+        rng, N, D = MersenneTwister(123456), 5, 3
+        x1 = randn(rng, N)
+        xD = ColVecs(randn(rng, D, N))
+        xD′ = RowVecs(randn(rng, N, D))
+
         foo_mean = x -> sum(abs2, x)
-        f = CustomMean(foo_mean)
+        m = CustomMean(foo_mean)
 
-        @test AbstractGPs._map_meanfunction(f, x) == map(foo_mean, x)
-        # differentiable_mean_function_tests(f, randn(rng, N), x)
+        for x in [x1, xD, xD′]
+            @test AbstractGPs._map_meanfunction(m, x) == map(foo_mean, x)
+            differentiable_mean_function_tests(m, randn(rng, N), x)
```

Comment:
Should we just call
```suggestion
            differentiable_mean_function_tests(rng, m, x)
```
instead (and remove the y = ...), or otherwise remove that (currently unused) method definition of differentiable_mean_function_tests?

Reply:
Seems like it would simplify things a bit, so I'm in favour.

Reply:
which one? remove the (rng, m, x) method, or apply this suggestion?

Reply:
I would be inclined to retain the method that just requires a `rng`, since we can now be confident that it will generate an appropriate tangent due to the call to `collect`.